src/main.rs

```diff
@@ -13,8 +16,29 @@ fn main() -> Result<(), std::io::Error> {
         io::stdin().read_line(&mut buffer)?;
 
         let tokenized = tokenize::tokenize(&mut buffer);
-        
-        if tokenized.cmd() == "cd" {
+
+        if let Some(full_alias_cmd) = aliases.get(tokenized.cmd()) {
+            let mut alias_parts: Vec<&str> = full_alias_cmd.split_whitespace().collect();
+            let alias_cmd = alias_parts.remove(0);
+            let alias_args = [&alias_parts[..], &tokenized.args()[..]].concat();
+
+            // Spawn the aliased command
+            let mut child = match Command::new(alias_cmd)
+                .args(&alias_args)
+                .spawn() {
+                    Ok(child) => child,
+                    Err(_) => {
+                        eprintln!("Failed to execute command");
+                        continue;
+                    }
+                };
+
+            // Wait for the command to finish
+            match child.wait() {
+                Ok(_) => {}
+                Err(e) => eprintln!("{}", e),
+            }
```

Comment:
Ah, I think you misunderstood what I meant with my previous comment.

Because lines 25-40 are just running a plain old process, let's move lines 21-23 outside of the big if/else statement, but just make sure that if there are no matches for the alias, we don't modify the command at all.

You can imagine a sort of pipeline of modifications that we perform before we actually run it. One stage of that pipeline is inserting the relevant aliases.